subsys/net/lib/prometheus/gauge.c

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/net/prometheus/gauge.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(pm_gauge, CONFIG_PROMETHEUS_LOG_LEVEL);
+
+void prometheus_gauge_set(struct prometheus_gauge *gauge, double value)
+{
+	if (value < 0) {
+		LOG_ERR("Invalid value");
+		return;
```

Comment:
How about making the logger return an error code?
```suggestion
		return -EINVAL;
```